jmclient/jmclient/wallet.py

```diff
@@ -566,6 +537,10 @@ def remove_old_utxos(self, tx):
         for (txid, index), val in ret.items():
             val['address'] = self.get_addr_path(val['path'])
             removed_utxos[hexlify(txid).decode('ascii') + ':' + str(index)] = val
+        if len(removed_utxos.keys()) > 0:
+            jlog.info('saw tx on network, removed_utxos=\n{}'.format('\n'.join(
+            '{} - {}'.format(u, fmt_tx_data(tx_data, self))
+            for u, tx_data in removed_utxos.items())))
```

Comment:
It seems a bit weird to put this in the wallet code. I'd more expect the caller to emit such a message.

Reply:
Indeed it does seem a bit weird, but the point is there isn't a caller, in a sense: since we're moving to a reactive mode for transactions hitting the wallet, this was not triggered by any action in Joinmarket code, only by what the blockchain told us. What is perhaps illogical is: we have this message for removals but not additions. On balance I think I'd prefer both. We could also put these messages in wallet service but it just seemed logical to keep this message (which already existed) but put it where it happens. What do you think?

Reply:
The thing is, this says something about network activity, but ultimately the wallet knows nothing about where its information is sourced from. So this log message should be somewhere closer to the actual data source (network info), which would be the `WalletService`.

Reply:
True, although it's a minor difference. I can put both added and removed as a debug msg in the `WalletService.transaction_monitor()` after `self.wallet.process_new_txs()`.

Reply:
Yes, I'd prefer it that way. (also: this would be the only `jlog` call in the whole wallet code)

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/359/commits/248dbbad63bd27862e7b3b9354468db13d09aec3